rslib/src/import_export/text/import.rs

```diff
@@ -40,6 +40,24 @@ struct Context<'a> {
     //progress: IncrementableProgress<ImportProgress>,
 }
 
+#[derive(Debug, Clone, Copy)]
+enum ImportResult {
+    Added,
+    Ignored,
+    Updated,
+}
+
+impl NoteLog {
+    fn log_result(&mut self, note: Note, result: ImportResult) {
+        match result {
+            ImportResult::Added => &mut self.new,
+            ImportResult::Ignored => &mut self.duplicate,
+            ImportResult::Updated => &mut self.updated,
```

Comment:
The messages generated from the log are currently apkg-specific. Not yet sure how to handle this.

Reply:
I'm not sure I'm following - do you mean importing.py:210+? What is specific to apkgs there?

Reply:
Sorry, it's confusing because I couldn't comment the actual lines of interest.
The problem is in `stringify_log()`, which uses messages like "Notes skipped, as they're already in your collection: { $val }". Compared to "First field matched: { $val }", which is used for CSV imports in the old code.
Maybe we need more different log queues like `full_duplicate` and `duplicate_first_field`.

Reply:
But I see, you found the relevant lines importing.py:210. Alternatively, we could use more generic messages, of course. E.g. "Added duplicate with first field: { $val }" would be just subsumed into "Notes added from file: { $val }".

Reply:
Good point, it would probably be better if we can preserve the mention of the first field. Separate log queues could work. One other alternative would be to add an enum field to ImportResponse that tells the GUI which of the two translation strings it should pick, eg 

```
  enum DuplicateMatchingStrategy {
    GUID = 0;
    FIRST_FIELD = 1;
  }
```

Might be a bit easier to implement the separate queues approach in the GUI though.

Reply:
I think this is note-specific, though. I a JSON file you could have both, notes with and without a guid/id. Then different queues are probably simpler.

Reply:
I ended up with something in between: a single queue for all first-field matches and an enum. It's not perfect, because the fluent template strings for CSV and apkg import are constructed differently, but it seemed too early to add new ones.